src/spec-node/containerFeatures.ts

```diff
@@ -294,6 +294,14 @@ async function getFeaturesBuildOptions(params: DockerResolverParameters, devCont
 		.replace('#{copyFeatureBuildStages}', getCopyFeatureBuildStages(featuresConfig, buildStageScripts))
 		.replace('#{devcontainerMetadata}', getDevcontainerMetadataLabel(imageMetadata, common.experimentalImageMetadata))
 		;
+	
+	const rawContainerEnv = imageMetadata.raw.find(entry => entry.containerEnv)?.containerEnv;
```

Comment:
Use `imageMetadata.config`, that has variables like `${remoteWorkspaceFolder}` replaced. (`imageMetadata.raw` comes without these variables replaced.)

Reply:
`imageMetadata` actually includes the features for which we already apply `containerEnv`. Can you try using `devContainerConfig.config.containerEnv`?

Reply:
> Can you try using devContainerConfig.config.containerEnv?

Works 👍